_docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme

```diff
@@ -86,113 +87,194 @@ function rhdp2_preprocess_node(&$variables) {
 
-      if ($author_url_alias === $current_alias) {
-        $variables['authors'] = [
-          '#markup' => reset($author_nodes)->getTitle(),
-        ];
+    // Prepares the authors render element.
+    if ($variables['view_mode'] === 'card') {
+      $authors = $node->get('field_author_evangelist')->getValue();
+      $tids = [];
+
+      foreach ($authors as $author) {
+        $tids[] = $author['target_id'];
       }
-      else {
-        // We can reset to retrieve the element since we've already verified there is only 1.
-        $authors_values = [
-          'name' => reset($author_nodes)->getTitle(),
-          // We need a Url object for Drupal, not a plain string.
-          'url' => Url::fromRoute('entity.node.canonical', ['node' => reset($author_nodes)->id()])
-        ];
-
-        // Get a renderable author Link and set the rhd-m-link class.
-        $authors_link = Link::fromTextAndUrl($authors_values['name'], $authors_values['url']);
-        $authors_link = $authors_link->toRenderable();
-        $authors_link['#attributes'] = ['class' => ['rhd-m-link']];
-
-        $variables['authors'] = $authors_link;
+
+      $author_nodes = Node::loadMultiple($tids);
+
+      if (count($author_nodes) < 1) {
+        $variables['authors'] = NULL;
       }
-    }
-    else {
-      $authors_values = [];
 
-      foreach ($author_nodes as $author_node) {
-        $authors_values[] = $author_node->getTitle();
+      if (count($author_nodes) === 1) {
+        $nid = reset($author_nodes)->id();
+        $author_url_alias = \Drupal::service('path.alias_manager')->getAliasByPath('/node/' . $nid);
+        $current_path = \Drupal::service('path.current')->getPath();
+        $current_alias = \Drupal::service('path.alias_manager')->getAliasByPath($current_path);
+
+        if ($author_url_alias === $current_alias) {
+          $variables['authors'] = [
+            '#markup' => reset($author_nodes)->getTitle(),
+          ];
+        }
+        else {
+          // Reset to retrieve the element since there is only 1.
+          $authors_values = [
+            'name' => reset($author_nodes)->getTitle(),
+            'url' => Url::fromRoute('entity.node.canonical', ['node' => reset($author_nodes)->id()]),
+          ];
+
+          // Get a renderable author Link and set the rhd-m-link class.
+          $authors_link = Link::fromTextAndUrl($authors_values['name'], $authors_values['url']);
+          $authors_link = $authors_link->toRenderable();
+          $authors_link['#attributes'] = ['class' => ['rhd-m-link']];
+
+          $variables['authors'] = $authors_link;
+        }
       }
+      else {
+        // @TODO: This still runs if the author_nodes is 0?
+        $authors_values = [];
 
-      $variables['authors'] = implode('&', $authors_values);
+        foreach ($author_nodes as $author_node) {
+          $authors_values[] = $author_node->getTitle();
+        }
+
+        $variables['authors'] = implode('&', $authors_values);
```

Comment:
Can we add space before and after the `&` in the implode()?

Reply:
This is done.